BuildScript.Nuke/BaseBuild.Compile.cs

```diff
@@ -95,9 +108,26 @@ private void CompileProject (IReadOnlyCollection<ProjectMetadata> projectFiles)
     );
   }
 
+  private string GetToolPath ()
+  {
+    var toolPath = MSBuildTasks.MSBuildPath;
+    var editions = new[] { "Enterprise", "Professional", "Community", "BuildTools", "Preview" };
+    if (!string.IsNullOrEmpty(MsBuildPath))
+      toolPath = MsBuildPath;
+    else if (!string.IsNullOrEmpty(MsBuildToolVersion))
+      toolPath = editions
+          .Select(
+              edition => Path.Combine(
+                  EnvironmentInfo.SpecialFolder(SpecialFolders.ProgramFilesX86)!,
+                  $@"Microsoft Visual Studio\2022\{edition}\MSBuild\{MsBuildToolVersion}\Bin\msbuild.exe"))
+          .First(File.Exists);
+    return toolPath;
+  }
```

Comment:
The support of the different toolversions isn't complete at the moment. Here, the path is always constructed for VS2022, please adapt to support the other VS versions aswell. 

The ToolVersion relates to the MSBuild version  bundled with VS:
- VS2015: 14.0
- VS2017: 15.0
- VS2019: Current
- VS2022: Current

Therefor the `MsBuildToolVersion` should be changed to `VisualStudioVersion` and parsed to return the fitting enum variant, the toolsversion can then be inferred from the vs version.

This code should have the same behavior as the resolver used in nuke, except for the added BuildTools2022 support.